README.md

```diff
@@ -1,2 +1,154 @@
+
+## Including other versions of Holochain
+
+By default the Holonix input references the `main` branch of the Holonix repository. Whatever the current revision of that branch at the moment of invoking the dev shell for the first time is will be written to the `flake.lock` file. The `main` branch uses version 0.4 of Holochain and compatible versions of scaffolding etc.
+
+To use other versions of Holochain in the dev shell, the `holonix` input must be modified.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main";
+    ...
+};
+```
+
+For Holochain v0.3, change it to `main-0.3`.
+
+```nix
+inputs = {
+    holonix.url = "github:holochain/holonix?ref=main-0.3";
+    ...
+};
+```
+
+Now running `nix develop` will update the flake lock file with the current revision of the `main-0.3` branch of the Holonix repository and enter the dev shell.
+
+## Defining packages to include in the dev shell
+
+It may be that you want to add or remove packages included in the dev shell. This is the relevant section
+
+```nix
+...
+packages = (with inputs'.holonix.packages; [
+    holochain
+    lair-keystore
+    hc-launch
+    hc-scaffold
+    hn-introspect
+    rust # For Rust development, with the WASM target included for zome builds
+])
+...
+```
+
+If you wanted to only use the Holochain binaries in a dev shell, all other packages can be removed from the default shell.
+
+```nix
+...
+packages = (with inputs'.holonix.packages; [
+    holochain
+    rust # For Rust development, with the WASM target included for zome builds
+])
+...
+```
+
+Other packages from the Nix packages collection can be added here.
+
+```nix
+]) ++ (with pkgs; [
+    nodejs_20 # For UI development
+    binaryen # For WASM optimisation
+    # Add any other packages you need here
+]);
+```
+
+For example the `jq` package can be added.
+
+```nix
+]) ++ (with pkgs; [
+    nodejs_20 # For UI development
+    binaryen # For WASM optimisation
+    # Add any other packages you need here
+    jq
+]);
+```
+
+> Note that by default Rust is included in the dev shell. If you want to use your globally installed version of Rust instead, remove the `rust` package from the list of packages.
+
+## Customized Holochain build
+
+In the default flake template, Holochain comes built with default features. If you need Holochain with a different feature set, you can customize the Holochain build. The `custom` template shows how to customize the Holochain build.
+
+To initialize a flake based on the custom template, run
+
+```shell
+nix flake init -t github:holochain/holonix#custom
+```
+
+In [the generated `flake.nix` file](./templates/custom/flake.nix), locate the section that specifies the cargo build parameters to be passed to the Holochain build.
+
+```nix
+let
+    # Enable CHC (chain head coordination) feature for Holochain package.
+    cargoExtraArgs = "--features chc";
```

Comment:
This has since been made a default feature, which makes this a bit confusing. Maybe showing how to build for wamr instead?